capstone/src/main/webapp/scripts/poll.js

```diff
@@ -34,40 +40,73 @@ function getPollOptions() {
       return pollData["votedOptions"];
     })
     .then(votedOptions => {
-      handleCheck(votedOptions);
+      if (votedOptions) {
+        handleCheck(votedOptions);
+      } else {
+        return;
+      }
     })
-    .then(checkWeek);
-}
-
-function checkWeek() {
-  getChallenge();
-  let now = new Date();
-  //Hard coding due date for now
-  let millisTillDueDate =
-    new Date(now.getFullYear(), now.getMonth(), now.getDate(), 13, 04, 0, 0) -
-    now;
-  if (millisTillDueDate >= 0) {
-    setTimeout(updatePoll, millisTillDueDate);
-  }
+    .then(getChallenge());
 }
 
+/**
+ * Gets the current challenge from server and post to DOM. Check if the challenge needs to be
+ * updated.
+ */
 function getChallenge() {
+  //Milliseconds until challenge due date.
+  let dueDateMillis = 0;
   fetch("challenge")
     .then(response => response.json())
     .then(challengeData => {
       const weeklyChallenge = document.getElementById("weekly-challenge");
-      weeklyChallenge.innerText = challengeData["challengeName"];
-    });
+      const dueDateContainer = document.getElementById("due-date");
+      if (challengeData) {
+        weeklyChallenge.innerText = challengeData["challengeName"];
+        dueDateMillis = challengeData["dueDate"];
+        const dueDate = new Date(dueDateMillis).toString();
+        dueDateContainer.innerText = `Due: ${dueDate}`;
+      } else {
+        noChallengeText();
+        updatePoll();
+      }
+    })
+    .then(() => checkWeek(dueDateMillis));
+}
+
+function noChallengeText() {
+  const weeklyChallenge = document.getElementById("weekly-challenge");
+  const dueDateContainer = document.getElementById("due-date");
+  weeklyChallenge.innerText = NO_CHALLENGES;
+  dueDateContainer.innerText = "";
+}
+
+/**
+ * Checks if the challenge needs to be updated.
+ * @param {long} dueDateMillis
+ */
+function checkWeek(dueDateMillis) {
+  let now = new Date();
+  let millisTillDueDate = new Date(dueDateMillis) - now;
+  if (millisTillDueDate < 0) {
+    updatePoll();
+  }
 }
 
+/**
+ * Deletes the top poll option, adding that option as a new challenge to the database.
+ */
 function updatePoll() {
-  fetch("delete-top-option", { method: "POST" }).then(addChallengeToDb);
+  fetch("delete-top-option", { method: "POST" }).then(addChallengeToDb());
```

Comment:
I think `.then(addChallengeToDb);` was correct, otherwise `addChallengeToDb()` would be called and then the result (`undefined`) is passed into then.

Reply:
Alternatively, `.then(() => addChallengeToDb())`, but this is not necessary

Reply:
so whenever I call a method in .then should I just exclude the parentheses?